cs/src/Management/TunnelManagementClient.cs

```diff
@@ -400,11 +400,10 @@ private static void ValidateHttpHandler(HttpMessageHandler httpHandler)
                     case HttpStatusCode.Forbidden:
                         var ex = new UnauthorizedAccessException(errorMessage, hrex);
 
-                        if (response.Headers.WwwAuthenticate?.Count > 0)
-                        {
-                            ex.SetAuthenticationSchemes(response.Headers.WwwAuthenticate);
-                        }
-
+                        // The HttpResponseHeaders.WwwAuthenticate property does not correctly
+                        // handle multiple values! Get the values by name instead.
+                        var authHeaderValues = response.Headers.GetValues("WWW-Authenticate");
```

Comment:
Nit: it can be `HeaderNames.WWWAuthenticate`

Reply:
`HeaderNames` is in the `Microsoft.Net.Http.Headers` assembly, which is not currently referenced by the SDK, and I don't think it's worth adding the dependency just for this.